src/dotnet/commands/dotnet-install/ProjectRestorer.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.DotNet.ExecutablePackageObtainer;
+using Microsoft.DotNet.PlatformAbstractions;
+using Microsoft.Extensions.EnvironmentAbstractions;
+
+namespace Microsoft.DotNet.Cli
+{
+    internal class ProjectRestorer : ICanRestoreProject
+    {
+        public void Restore(
+            FilePath tempProjectPath,
+            DirectoryPath assetJsonOutput,
+            FilePath? nugetconfig)
+        {
+            var argsToPassToRestore = new List<string>();
```

Comment:
If the order of the options doesn't matter, could we initialize this to the `tempProjectPath`, the `--runtime` option, and the `-p` option, then add `--configfile` conditionally?

Reply:
I tend to think list with initialization will not be append later. So I keep an empty list in the beginning.